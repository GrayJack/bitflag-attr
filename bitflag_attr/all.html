<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="List of all items in this crate"><title>List of all items in this crate</title><script>if(window.location.protocol!=="file:")document.head.insertAdjacentHTML("beforeend","SourceSerif4-Regular-6b053e98.ttf.woff2,FiraSans-Italic-81dc35de.woff2,FiraSans-Regular-0fe48ade.woff2,FiraSans-MediumItalic-ccf7e434.woff2,FiraSans-Medium-e1aa3f0a.woff2,SourceCodePro-Regular-8badfe75.ttf.woff2,SourceCodePro-Semibold-aa29a496.ttf.woff2".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/${f}">`).join(""))</script><link rel="stylesheet" href="../static.files/normalize-9960930a.css"><link rel="stylesheet" href="../static.files/rustdoc-cf3c48c1.css"><meta name="rustdoc-vars" data-root-path="../" data-static-root-path="../static.files/" data-current-crate="bitflag_attr" data-themes="" data-resource-suffix="" data-rustdoc-version="1.86.0-nightly (92bedea1c 2025-02-11)" data-channel="nightly" data-search-js="search-581efc7a.js" data-settings-js="settings-6dad6058.js" ><script src="../static.files/storage-3a5871a4.js"></script><script defer src="../static.files/main-4d63596a.js"></script><noscript><link rel="stylesheet" href="../static.files/noscript-893ab5e7.css"></noscript><link rel="alternate icon" type="image/png" href="../static.files/favicon-32x32-6580c154.png"><link rel="icon" type="image/svg+xml" href="../static.files/favicon-044be391.svg"></head><body class="rustdoc mod sys"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle" title="show sidebar"></button></nav><nav class="sidebar"><div class="sidebar-crate"><h2><a href="../bitflag_attr/index.html">bitflag_<wbr>attr</a><span class="version">0.11.0</span></h2></div><div class="sidebar-elems"><section id="rustdoc-toc"><h3><a href="#macros">Crate Items</a></h3><ul class="block"><li><a href="#macros" title="Macros">Macros</a></li><li><a href="#structs" title="Structs">Structs</a></li><li><a href="#traits" title="Traits">Traits</a></li><li><a href="#functions" title="Functions">Functions</a></li><li><a href="#attributes" title="Attribute Macros">Attribute Macros</a></li></ul></section><div id="rustdoc-modnav"></div></div></nav><div class="sidebar-resizer"></div><main><div class="width-limiter"><rustdoc-search></rustdoc-search><section id="main-content" class="content"><h1>List of all items</h1><h3 id="structs">Structs</h3><ul class="all-items"><li><a href="example_generated/struct.ExampleFlags.html">example_generated::ExampleFlags</a></li><li><a href="iter/struct.Iter.html">iter::Iter</a></li><li><a href="iter/struct.IterNames.html">iter::IterNames</a></li><li><a href="parser/struct.ParseError.html">parser::ParseError</a></li></ul><h3 id="traits">Traits</h3><ul class="all-items"><li><a href="trait.BitsPrimitive.html">BitsPrimitive</a></li><li><a href="trait.Flags.html">Flags</a></li><li><a href="parser/trait.ParseHex.html">parser::ParseHex</a></li></ul><h3 id="macros">Macros</h3><ul class="all-items"><li><a href="macro.bitflag_match.html">bitflag_match</a></li></ul><h3 id="attributes">Attribute Macros</h3><ul class="all-items"><li><a href="attr.bitflag.html">bitflag</a></li></ul><h3 id="functions">Functions</h3><ul class="all-items"><li><a href="parser/fn.from_text.html">parser::from_text</a></li><li><a href="parser/fn.from_text_strict.html">parser::from_text_strict</a></li><li><a href="parser/fn.from_text_truncate.html">parser::from_text_truncate</a></li><li><a href="parser/fn.to_writer.html">parser::to_writer</a></li><li><a href="parser/fn.to_writer_strict.html">parser::to_writer_strict</a></li><li><a href="parser/fn.to_writer_truncate.html">parser::to_writer_truncate</a></li></ul></section></div></main></body></html>