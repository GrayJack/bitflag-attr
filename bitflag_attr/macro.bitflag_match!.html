<!DOCTYPE html>
<html lang="en">
<head>
    <meta http-equiv="refresh" content="0;URL=macro.bitflag_match.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="macro.bitflag_match.html">macro.bitflag_match.html</a>...</p>
    <script>location.replace("macro.bitflag_match.html" + location.search + location.hash);</script>
</body>
</html>