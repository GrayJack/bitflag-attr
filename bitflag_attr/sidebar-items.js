window.SIDEBAR_ITEMS = {"attr":["bitflag"],"macro":["bitflag_match"],"mod":["example_generated","iter","parser"],"trait":["BitsPrimitive","Flags"]};