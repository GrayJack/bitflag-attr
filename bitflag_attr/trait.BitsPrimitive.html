<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="Primitive types that can be used with `bitflag` attribute implement this trait."><title>BitsPrimitive in bitflag_attr - Rust</title><script>if(window.location.protocol!=="file:")document.head.insertAdjacentHTML("beforeend","SourceSerif4-Regular-6b053e98.ttf.woff2,FiraSans-Italic-81dc35de.woff2,FiraSans-Regular-0fe48ade.woff2,FiraSans-MediumItalic-ccf7e434.woff2,FiraSans-Medium-e1aa3f0a.woff2,SourceCodePro-Regular-8badfe75.ttf.woff2,SourceCodePro-Semibold-aa29a496.ttf.woff2".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/${f}">`).join(""))</script><link rel="stylesheet" href="../static.files/normalize-9960930a.css"><link rel="stylesheet" href="../static.files/rustdoc-cf3c48c1.css"><meta name="rustdoc-vars" data-root-path="../" data-static-root-path="../static.files/" data-current-crate="bitflag_attr" data-themes="" data-resource-suffix="" data-rustdoc-version="1.86.0-nightly (92bedea1c 2025-02-11)" data-channel="nightly" data-search-js="search-581efc7a.js" data-settings-js="settings-6dad6058.js" ><script src="../static.files/storage-3a5871a4.js"></script><script defer src="sidebar-items.js"></script><script defer src="../static.files/main-4d63596a.js"></script><noscript><link rel="stylesheet" href="../static.files/noscript-893ab5e7.css"></noscript><link rel="alternate icon" type="image/png" href="../static.files/favicon-32x32-6580c154.png"><link rel="icon" type="image/svg+xml" href="../static.files/favicon-044be391.svg"></head><body class="rustdoc trait"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle" title="show sidebar"></button></nav><nav class="sidebar"><div class="sidebar-crate"><h2><a href="../bitflag_attr/index.html">bitflag_<wbr>attr</a><span class="version">0.11.0</span></h2></div><div class="sidebar-elems"><section id="rustdoc-toc"><h2 class="location"><a href="#">Bits<wbr>Primitive</a></h2><h3><a href="#required-associated-consts">Required Associated Constants</a></h3><ul class="block"><li><a href="#associatedconstant.ALL" title="ALL">ALL</a></li><li><a href="#associatedconstant.EMPTY" title="EMPTY">EMPTY</a></li></ul><h3><a href="#foreign-impls">Implementations on Foreign Types</a></h3><ul class="block"><li><a href="#impl-BitsPrimitive-for-i128" title="i128">i128</a></li><li><a href="#impl-BitsPrimitive-for-i16" title="i16">i16</a></li><li><a href="#impl-BitsPrimitive-for-i32" title="i32">i32</a></li><li><a href="#impl-BitsPrimitive-for-i64" title="i64">i64</a></li><li><a href="#impl-BitsPrimitive-for-i8" title="i8">i8</a></li><li><a href="#impl-BitsPrimitive-for-isize" title="isize">isize</a></li><li><a href="#impl-BitsPrimitive-for-u128" title="u128">u128</a></li><li><a href="#impl-BitsPrimitive-for-u16" title="u16">u16</a></li><li><a href="#impl-BitsPrimitive-for-u32" title="u32">u32</a></li><li><a href="#impl-BitsPrimitive-for-u64" title="u64">u64</a></li><li><a href="#impl-BitsPrimitive-for-u8" title="u8">u8</a></li><li><a href="#impl-BitsPrimitive-for-usize" title="usize">usize</a></li></ul><h3><a href="#dyn-compatibility">Dyn Compatibility</a></h3><h3><a href="#implementors">Implementors</a></h3></section><div id="rustdoc-modnav"><h2 class="in-crate"><a href="index.html">In crate bitflag_<wbr>attr</a></h2></div></div></nav><div class="sidebar-resizer"></div><main><div class="width-limiter"><rustdoc-search></rustdoc-search><section id="main-content" class="content"><div class="main-heading"><div class="rustdoc-breadcrumbs"><a href="index.html">bitflag_attr</a></div><h1>Trait <span class="trait">BitsPrimitive</span><button id="copy-path" title="Copy item path to clipboard">Copy item path</button></h1><rustdoc-toolbar></rustdoc-toolbar><span class="sub-heading"><a class="src" href="../src/bitflag_attr/lib.rs.html#273-296">Source</a> </span></div><pre class="rust item-decl"><code>pub trait BitsPrimitive:
    Sealed
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/marker/trait.Copy.html" title="trait core::marker::Copy">Copy</a>
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/cmp/trait.PartialEq.html" title="trait core::cmp::PartialEq">PartialEq</a>
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/ops/bit/trait.BitAnd.html" title="trait core::ops::bit::BitAnd">BitAnd</a>&lt;Output = Self&gt;
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/ops/bit/trait.BitOr.html" title="trait core::ops::bit::BitOr">BitOr</a>&lt;Output = Self&gt;
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/ops/bit/trait.BitXor.html" title="trait core::ops::bit::BitXor">BitXor</a>&lt;Output = Self&gt;
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/ops/bit/trait.Not.html" title="trait core::ops::bit::Not">Not</a>&lt;Output = Self&gt;
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/ops/bit/trait.BitAndAssign.html" title="trait core::ops::bit::BitAndAssign">BitAndAssign</a>
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/ops/bit/trait.BitOrAssign.html" title="trait core::ops::bit::BitOrAssign">BitOrAssign</a>
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/ops/bit/trait.BitXorAssign.html" title="trait core::ops::bit::BitXorAssign">BitXorAssign</a>
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/fmt/trait.Binary.html" title="trait core::fmt::Binary">Binary</a>
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/fmt/trait.LowerHex.html" title="trait core::fmt::LowerHex">LowerHex</a>
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/fmt/trait.UpperHex.html" title="trait core::fmt::UpperHex">UpperHex</a>
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/fmt/trait.Octal.html" title="trait core::fmt::Octal">Octal</a>
    + <a class="trait" href="https://doc.rust-lang.org/nightly/core/marker/trait.Sized.html" title="trait core::marker::Sized">Sized</a>
    + 'static {
    const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self;
    const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self;
}</code></pre><details class="toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>Primitive types that can be used with <a href="attr.bitflag.html" title="attr bitflag_attr::bitflag"><code>bitflag</code></a> attribute implement this trait.</p>
</div></details><h2 id="required-associated-consts" class="section-header">Required Associated Constants<a href="#required-associated-consts" class="anchor">§</a></h2><div class="methods"><details class="toggle" open><summary><section id="associatedconstant.EMPTY" class="method"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#292">Source</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self</h4></section></summary><div class="docblock"><p>A value with all bits unset.</p>
</div></details><details class="toggle" open><summary><section id="associatedconstant.ALL" class="method"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#295">Source</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self</h4></section></summary><div class="docblock"><p>A value with all bits set.</p>
</div></details></div><h2 id="dyn-compatibility" class="section-header">Dyn Compatibility<a href="#dyn-compatibility" class="anchor">§</a></h2><div class="dyn-compatibility-info"><p>This trait is <b>not</b> <a href="https://doc.rust-lang.org/nightly/reference/items/traits.html#dyn-compatibility">dyn compatible</a>.</p><p><i>In older versions of Rust, dyn compatibility was called "object safety", so this trait is not object safe.</i></p></div><h2 id="foreign-impls" class="section-header">Implementations on Foreign Types<a href="#foreign-impls" class="anchor">§</a></h2><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-i8" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#impl-BitsPrimitive-for-i8" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.i8.html">i8</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-1" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.EMPTY-1" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0i8</h4></section><section id="associatedconstant.ALL-1" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.ALL-1" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = -1i8</h4></section></div></details><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-i16" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#impl-BitsPrimitive-for-i16" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.i16.html">i16</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-2" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.EMPTY-2" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0i16</h4></section><section id="associatedconstant.ALL-2" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.ALL-2" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = -1i16</h4></section></div></details><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-i32" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#impl-BitsPrimitive-for-i32" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.i32.html">i32</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-3" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.EMPTY-3" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0i32</h4></section><section id="associatedconstant.ALL-3" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.ALL-3" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = -1i32</h4></section></div></details><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-i64" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#impl-BitsPrimitive-for-i64" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.i64.html">i64</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-4" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.EMPTY-4" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0i64</h4></section><section id="associatedconstant.ALL-4" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.ALL-4" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = -1i64</h4></section></div></details><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-i128" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#impl-BitsPrimitive-for-i128" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.i128.html">i128</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-5" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.EMPTY-5" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0i128</h4></section><section id="associatedconstant.ALL-5" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.ALL-5" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = -1i128</h4></section></div></details><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-isize" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#impl-BitsPrimitive-for-isize" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.isize.html">isize</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-6" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.EMPTY-6" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0isize</h4></section><section id="associatedconstant.ALL-6" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#323">Source</a><a href="#associatedconstant.ALL-6" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = -1isize</h4></section></div></details><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-u8" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#impl-BitsPrimitive-for-u8" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.u8.html">u8</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-7" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.EMPTY-7" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0u8</h4></section><section id="associatedconstant.ALL-7" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.ALL-7" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = 255u8</h4></section></div></details><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-u16" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#impl-BitsPrimitive-for-u16" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.u16.html">u16</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-8" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.EMPTY-8" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0u16</h4></section><section id="associatedconstant.ALL-8" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.ALL-8" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = 65_535u16</h4></section></div></details><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-u32" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#impl-BitsPrimitive-for-u32" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.u32.html">u32</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-9" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.EMPTY-9" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0u32</h4></section><section id="associatedconstant.ALL-9" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.ALL-9" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = 4_294_967_295u32</h4></section></div></details><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-u64" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#impl-BitsPrimitive-for-u64" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.u64.html">u64</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-10" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.EMPTY-10" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0u64</h4></section><section id="associatedconstant.ALL-10" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.ALL-10" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = 18_446_744_073_709_551_615u64</h4></section></div></details><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-u128" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#impl-BitsPrimitive-for-u128" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.u128.html">u128</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-11" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.EMPTY-11" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0u128</h4></section><section id="associatedconstant.ALL-11" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.ALL-11" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = 340_282_366_920_938_463_463_374_607_431_768_211_455u128</h4></section></div></details><details class="toggle implementors-toggle"><summary><section id="impl-BitsPrimitive-for-usize" class="impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#impl-BitsPrimitive-for-usize" class="anchor">§</a><h3 class="code-header">impl <a class="trait" href="trait.BitsPrimitive.html" title="trait bitflag_attr::BitsPrimitive">BitsPrimitive</a> for <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.usize.html">usize</a></h3></section></summary><div class="impl-items"><section id="associatedconstant.EMPTY-12" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.EMPTY-12" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.EMPTY" class="constant">EMPTY</a>: Self = 0usize</h4></section><section id="associatedconstant.ALL-12" class="associatedconstant trait-impl"><a class="src rightside" href="../src/bitflag_attr/lib.rs.html#324">Source</a><a href="#associatedconstant.ALL-12" class="anchor">§</a><h4 class="code-header">const <a href="#associatedconstant.ALL" class="constant">ALL</a>: Self = 18_446_744_073_709_551_615usize</h4></section></div></details><h2 id="implementors" class="section-header">Implementors<a href="#implementors" class="anchor">§</a></h2><div id="implementors-list"></div><script src="../trait.impl/bitflag_attr/trait.BitsPrimitive.js" data-ignore-extern-crates="std" async></script></section></div></main></body></html>