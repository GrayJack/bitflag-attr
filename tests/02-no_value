use bitflag_attr::bitflag;

#[bitflag(u8)]
#[derive(Debug, Clone, Copy, PartialEq, PartialOrd, Eq, Ord, Hash)]
pub enum Test {
    Flag1 = 1 << 0,
    Flag2,
}

fn main() {}
