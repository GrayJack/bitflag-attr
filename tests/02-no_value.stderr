error: a discriminant must be defined
 --> tests/02-no_value:7:5
  |
7 |     Flag2,
  |     ^^^^^
